frontend/src/components/Home.vue

```diff
@@ -0,0 +1,166 @@
+        <!-- table containe flists -->
+        <v-data-table density="compact"
+          :items="filteredFlist"
+          :headers="tableHeader"
+          dense
+          
+          class="thick-border "
+          items-per-page="25"  
+        >
+        <template #item.name="{ value }">
+              <v-icon icon="mdi-text-box" class="mr-1"  color="grey"/>
+              <span class="file-name">{{ value }}</span>
+        </template>
+        <template v-slot:item.preview = "{index}" >
+          <a :href="`/` + filteredFlist[index].path_uri">
+            <v-btn class="elevation-0">
+                  <v-icon icon="mdi-eye-outline" color="grey"></v-icon>
+            </v-btn>
+          </a>
+        </template>
+        <template #item.size="{value}">
+          {{filesize(value, {standard: "jedec", precision: 3})}}
+        </template>
+          <template #item.last_modified="{ value }">
+            {{ new Date(value * 1000).toString().split("(")[0] }}
+          </template>
+          <template #item.path_uri="{ value }">
+            <v-btn class="elevation-0">
+              <a :href="baseURL + `/` + value" download>
+                <v-icon icon="mdi-download" color="grey"></v-icon
+              ></a>
+              <v-tooltip activator="parent" location="start"
+                >Download flist</v-tooltip
+              >
+            </v-btn>
+            <v-btn @click="copyLink(baseURL + `/` + value)" class="elevation-0">
+              <v-icon icon="mdi-content-copy" color="grey"></v-icon>
+              <v-tooltip activator="parent">Copy Link</v-tooltip>
+            </v-btn>
+          </template>
+        </v-data-table>
+      </v-container>
+    </v-main>
+    <Footer></Footer>
+  </v-app>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref, watch } from "vue";
+import Navbar from "./Navbar.vue";
+import Footer from "./Footer.vue";
+import image from "../assets/home.png";
+import { FlistsResponseInterface, FlistBody } from "../types/Flist.ts";
+import { toast } from "vue3-toastify";
+import "vue3-toastify/dist/index.css";
+import { api } from "../client.ts";
+import { copyLink } from "../helpers.ts";
+import {filesize} from "filesize";
+
+const baseURL = import.meta.env.VITE_API_URL;
+
+
+const tableHeader = [
+  { title: "File Name", key: "name" },
+  { title: "Preview", key:"preview"},
+  { title: "Size", key: "size" },
+  { title: "Last Modified", key: "last_modified" },
+  { title: "Download", key: "path_uri", sortable: false },
+];
+var flists = ref<FlistsResponseInterface>({});
+const username = ref("");
+const userNameList = ref<string[]>([]);
+let filteredFlist = ref<FlistBody[]>([]);
+const filteredFlistFn = () => {
+  filteredFlist.value = [];
+  const map = flists.value;
+  if (username.value.length === 0) {
+    for (var flistMap in map) {
+      for (let flist of map[flistMap]) {
+        if (flist.progress === 100) {
+          filteredFlist.value.push(flist);
+        }
+      }
+    }
```

Comment:
Instead of this, you can use filteredFlist.value = Array.from(map, ([_, value]) => value).

Reply:
Progress bar is broken.

![image](https://github.com/user-attachments/assets/1a824ff8-638b-4813-8098-bba22c3c276e)

Reply:
When I click preview, it redirects to another page. The preview should be done in the same page in a dialogue.

Reply:
Download & progress value didn't get changed until I refreshed the page.

Reply:
this page does not poll on the job so it needs to be refreshed